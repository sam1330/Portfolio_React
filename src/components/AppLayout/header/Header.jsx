import React from "react";
import { useTranslation } from "react-i18next";

import "./header.scss";

const Header = () => {

  const [t, i18n] = useTranslation();

  return (
    <nav className="header">
      <a href="#about-me" className="menu-item">
        {t("ABOUT_ME")}
      </a>
      <a href="#" className="menu-item">
        {t("TECHNOLOGIES")}
      </a>
      <a href="#" className="menu-item">
        {t("PROJECTS")}
      </a>
      <a href="#" className="menu-item">
        {t("CONTACT")}
      </a>
    </nav>
  );
};

export default Header;
