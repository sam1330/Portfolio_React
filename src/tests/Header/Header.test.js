import React from "react"
import { mount } from "enzyme";
import Header from "../../components/AppLayout/header/Header";

describe("Header Tests", () => {
  const wrapper = mount(<Header />);
  
  test('should render <Header />', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render correct translation', () => {
    expect(wrapper.find('a').at(0).text().trim()).toBe("ABOUT_ME");
  });
});