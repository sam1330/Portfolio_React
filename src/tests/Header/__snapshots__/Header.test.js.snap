// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Tests should render <Header /> 1`] = `
<nav
  className="header"
>
  <a
    className="menu-item"
    href="#about-me"
  >
    ABOUT_ME
  </a>
  <a
    className="menu-item"
    href="#"
  >
    TECHNOLOGIES
  </a>
  <a
    className="menu-item"
    href="#"
  >
    PROJECTS
  </a>
  <a
    className="menu-item"
    href="#"
  >
    CONTACT
  </a>
</nav>
`;
